File: 4 Kyu/4 Kyu - Factorial tail.c
#define MIN(a,b) (((a) < (b)) ? (a) : (b))

int zeroes (int base, int number) { // to fix //

    int i,p,c,zeroes = number,k;


      for (i = 2; i <= base; i++) {
          if (base % i == 0) {

             p = 0;
             while (base % i == 0) {
                   p++;
                   base /= i;
             }

             c = 0;
             k = number;
             while (k/i > 0) {
                   c += k/i;
                   k /= i;
             }

          zeroes = MIN(zeroes,c/p);
          }
      }

  return zeroes;
}

File: 4 Kyu/4 kyu - Factorial tail.rs
fn zeroes(base: i32, number: i32) -> i32 {
    let mut fact = base;
    let mut zeroes = number;
    
    for i in 2..= fact {
        if fact % i == 0 {
            
            let mut p = 0;
            while fact % i == 0 {
                p += 1;
                fact /= i;
            }
            
            let mut c = 0;
            let mut k = number;
            while k / i > 0 {
                c += k / i;
                k /= i;
            }
            
            zeroes = std::cmp::min (zeroes, c / p);
        }
    }
    
    zeroes
}
